libpod/container_internal_linux.go

```diff
@@ -1430,46 +1431,55 @@ func (c *Container) generatePasswd() (string, error) {
 		groupspec string
 		gid       int
 	)
-	if c.config.User == "" {
+	if !c.config.UserNsKeepId && c.config.User == "" {
 		return "", nil
 	}
-	splitSpec := strings.SplitN(c.config.User, ":", 2)
-	userspec := splitSpec[0]
-	if len(splitSpec) > 1 {
-		groupspec = splitSpec[1]
-	}
-	// If a non numeric User, then don't generate passwd
-	uid, err := strconv.ParseUint(userspec, 10, 32)
-	if err != nil {
-		return "", nil
-	}
-	// Lookup the user to see if it exists in the container image
-	_, err = lookup.GetUser(c.state.Mountpoint, userspec)
-	if err != nil && err != user.ErrNoPasswdEntries {
-		return "", err
-	}
-	if err == nil {
-		return "", nil
+	originPasswdFile := filepath.Join(c.state.Mountpoint, "/etc/passwd")
```

Comment:
I think it's time to either break this code into a separate function or move it into a `pkg/passwd`.

It was already really hard to brain-parse the code before but now we have four nested branches that let my brain parser overflow.

Breaking the code into a separate function would simplify it (as we can use returns), we could add unit tests and ideally add more comments.

Reply:
Done